spec/ddtrace/contrib/grape/tracer_spec.rb

```diff
@@ -333,79 +361,79 @@
+        expect(render_span).to have_error_message('Ouch!')
+        expect(render_span.get_tag('error.stack')).to include('grape/tracer_spec.rb')
+        expect(render_span.parent).to eq(run_span)
 
-          expect(run_span.name).to eq('grape.endpoint_run')
-          expect(run_span.span_type).to eq('web')
-          expect(run_span.service).to eq('grape')
-          expect(run_span.resource).to eq('RackTestingAPI#hard_failure')
-          expect(run_span).to have_error
-          expect(run_span.parent).to eq(rack_span)
-
-          expect(rack_span.name).to eq('rack.request')
-          expect(rack_span.span_type).to eq('web')
-          expect(rack_span.service).to eq('rack')
-          expect(rack_span.resource).to eq('RackTestingAPI#hard_failure')
-          expect(rack_span).to have_error
-          expect(rack_span.parent).to be_nil
-        end
+        expect(run_span.name).to eq('grape.endpoint_run')
+        expect(run_span.span_type).to eq('web')
+        expect(run_span.service).to eq('grape')
+        expect(run_span.resource).to eq('RackTestingAPI#hard_failure')
+        expect(run_span).to have_error
+        expect(run_span.parent).to eq(rack_span)
+
+        expect(rack_span.name).to eq('rack.request')
+        expect(rack_span.span_type).to eq('web')
+        expect(rack_span.service).to eq('rack')
+        expect(rack_span.resource).to eq('RackTestingAPI#hard_failure')
+        expect(rack_span).to have_error
+        expect(rack_span.parent).to be_nil
       end
+    end
 
-      context 'missing route' do
-        subject(:response) { get '/api/not_existing' }
+    context 'missing route' do
+      subject(:response) { get '/api/not_existing' }
 
-        it_behaves_like 'measured span for integration', true do
-          before do
-            expect(subject.status).to eq(404)
-          end
+      it_behaves_like 'measured span for integration', true do
+        before do
+          expect(subject.status).to eq(404)
         end
+      end
 
-        it 'it should not impact the Rack integration that must work as usual' do
-          expect(subject.status).to eq(404)
-          expect(spans.length).to eq(1)
+      it 'it should not impact the Rack integration that must work as usual' do
+        expect(subject.status).to eq(404)
+        expect(spans.length).to eq(1)
 
-          rack_span = spans[0]
+        rack_span = spans[0]
 
-          expect(rack_span.name).to eq('rack.request')
-          expect(rack_span.span_type).to eq('web')
-          expect(rack_span.service).to eq('rack')
-          expect(rack_span.resource).to eq('GET 404')
-          expect(rack_span).to_not have_error
-          expect(rack_span.parent).to be_nil
-        end
+        expect(rack_span.name).to eq('rack.request')
+        expect(rack_span.span_type).to eq('web')
+        expect(rack_span.service).to eq('rack')
+        expect(rack_span.resource).to eq('GET 404')
+        expect(rack_span).to_not have_error
```

Comment:
Something we'll have to be concious of when refactoring this to `error_statuses` is that we'll need to figure out if the top level rack spans should not record the error either

Reply:
Rack only sets it as error in [case of 5XXs](https://github.com/DataDog/dd-trace-rb/blob/7f608a861a9725930fbc8210d42921db30559ae1/lib/ddtrace/contrib/rack/middlewares.rb#L202-L204).

I think we should implement grape to our best, using `error_statuses` as we should and then apply this same pattern to `rack` later.

If we try to also address Rack in this PR it will become too complex.